gui: move network calls to a dedicated process

benoit-pierre opened the pull request:
To avoid freezing the GUI.

Notes:
- the worker process is forked, but a new interpreter is then started to ensure a clean slate: I could not find a way to dynamically require `Gtk et al.` only in the main process.
- the worker process higher priority requests first (FIFO order).
- the worker still process only one request at a time, so if for example the Invidious instance is down, a timeout of the current request can take a while and delay other (including higher priority) requests.
- the thumbnails are now independently saved to disk in `~/.cache/pipe-viewer`.
- ~the results list is now cleared **before** sending a new search request (instead of before displaying the results), rational: there's no much point in being able to keep interacting with the current results as most search request should be fast.~

TODO:
- [x] get rid of the couple of remaining network requests in the main process (when saving a channel).
- [x] get rid of the `WWW::PipeViewer` instance (`$yv_obj`) in the main process? Moving some of the helper methods still used to `WWW::PipeViewer::Utils` would make the separation between the part that needs online access and the offline stuff clearer: only calls left are to `video_categories` (return an hard-coded list, no network access), and `proxy_(exec|system)` (which uses the internal `LWP::UserAgent` instance for forwarding the proxy settings).
- [x] get rid of the annoying details window layout shift when filling the extra info: mostly addressed by #121, there's still some minor shifting due to inconsistencies in the way dates are handled.

Thoughts?

benoit-pierre commented:
Any feedback on this?

benoit-pierre commented:
OK, I think this is ready for review.

trizen commented on bin/gtk-pipe-viewer:
```diff
@@ -1744,19 +1866,9 @@ sub show_comments_window {
-        [],
-        Glib::G_PRIORITY_LOW
-    );
+    set_comments;
```
We should prefer calling functions explicitly with `()`. Mainly because it's more readable and also because the code won't break if the function declaration is moved after the function call.

Example:

```perl
use 5.014;

test;        # error
#test();     # this works

sub test {
    say "here";
}
```

trizen reviewed (changes requested):
Very impressive work!

benoit-pierre commented:
OK, I have addressed you comments.

I've also made a few additional changes:
- added some progress spinners to the comments and details dialogs.
- tweaked the worker debug traces to only show at higher level (2 and above), as it gets pretty chatty.
- and a few small cleanup.
